tfhe/src/core_crypto/gpu/vec.rs

```diff
@@ -23,30 +24,72 @@ use tfhe_cuda_backend::cuda_bind::{
 /// memory, it is pretty close to a `Vec`. That being said, it only present a very very limited api.
 #[derive(Debug)]
 pub struct CudaVec<T: Numeric> {
-    ptr: CudaPtrMut,
-    len: usize,
-    gpu_index: u32,
+    pub ptr: Vec<*mut c_void>,
+    pub lengths: Vec<usize>,
+    pub gpu_indexes: Vec<u32>,
     _phantom: PhantomData<T>,
```

Comment:
So I get the point of having the CudaVec like this, it allows to easily have the same data (keyswitchkey, bootstrapping key) on all GPUs easily, but since you allow differents lengths it means that each data on each GPU could be totally different, won't it be too easy to misuse ?

Reply:
Actually I'm not sure yet, but it's possible that in the end we will copy the ciphertexts to different GPUs. Since the number of GPUs does not necessarily divide the number of blocks, we would end up having a different work load on each GPU, and hence a different length for each. That's why I defined it this way.